assets/src/components/map.tsx

```diff
@@ -265,6 +268,51 @@ export const autoCenter = (
   }
 }
 
+const StreetViewControl = ({
+  streetViewEnabled: streetViewEnabled,
+  setStreetViewEnabled: setStreetViewEnabled,
+}: StreetViewControlProps): JSX.Element | null => {
+  const map = useMap()
+  const portalParent = map
+    .getContainer()
+    .querySelector(".leaflet-control-container")
+  const [portalElement, setPortalElement] = useState<HTMLElement | null>(null)
+  const id = useId()
+
+  useEffect(() => {
+    if (!portalParent || !portalElement) {
+      setPortalElement(document.createElement("div"))
+    }
+
+    if (portalParent && portalElement) {
+      portalElement.className =
+        "leaflet-control leaflet-bar m-vehicle-map__street-view-control"
+      portalParent.append(portalElement)
+      Leaflet.DomEvent.disableClickPropagation(portalElement)
+    }
+
+    return () => portalElement?.remove()
+  }, [portalElement, portalParent, setStreetViewEnabled])
+
+  const control = (
+    <>
+      <span id={id}>Street View</span>
+      <div className="form-check form-switch">
```

Comment:
suggestion/question: can a [`<label for=[id]>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attr-for) be used here instead of `<span>`/`aria-labelledby={id}`?

I also think that the ID should be declaritive/a function input, but I'm not opposed to `useId`.

Reply:
> but I still don't want to do things that will cause multiple instantiations of a single component to step on each other in weird ways.

Agreed.
I think what I'd prefer is generating an id but prefixing it with the component name or something, but this is because I don't like opaque web-apps that try to do "security through obscurity" and I think that it would be a better ergonomics to include some human readable information in there (https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id there's a warning callout on the page that sorta touches on this idea)

This is also assuming that prefixing the result of `useId` doesn't break anything.

Reply:
Ah, I see. Yes, I like the idea of prefixing it with something and it doesn't seem to break anything, I'll go ahead and do that.

Reply:
One thing to flag re: using `<label>` is that the contents of the label tag become clickable and effectively fire a click for the element they are labeling. I think that's probably fine if it's standard behavior but it's worth noting.